### ISSUE TITLE
Webapp - Build Models

### ISSUE THREAD

**Issue by drkohlipk** [drkohlipk/SDVC#5]

**Comment by TheOnlyRealTodd** [comment]
Veteran Model
-------------------------
- VeteranId (integer)
- FirstName (string)
- LastName (string)
- Branch of Service/Department served in (string) (Can be a dropdown menu on the UI side)
- Email (string) w/email restriction
- Username (We could make their email _**be**_ their username if you want.) w/ input validation for username and duplicates (string)
- Password (validation for password requirements) (string)

^^^^Those are the absolute necessities, however we can also have these included in the model if you want and also possibly we could query the already-existing paper database to populate the below:

- Address1 (string)
- Address2 (string)
- City (string)
- State (string) (Can be a dropdown menu on UI side)
- Zip (or postal) (string due to possibility of non-numeric chars)
- Country (If we plan on addresses being out-of-country of course. (string)
- HomePhone (string)
- CellPhone (string)

Add any input needed. Please note the separation of fields, this is for database integrity. I.E. Combining the address street, city, and state into one line is a no-no.

**Comment by drkohlipk** [comment]
Agreed, will also need a FK for a tie into services used (read VSO), for tracking purposes.

**Comment by stevengum** [comment]
Just hazarding a guess, but we should make the second half of fields be optional in relation to the upper half. If some potential users are in-between places, requiring an address could exclude these persons. Building on top of this, once we see a sample of the SDVC's dataset we can determine if including mailing addresses is something to consider implementing (I assumed address2 was for Apt, Suite etc, correct me if I'm wrong here).

We might consider adding a phone number field to the required list and allowing users to indicate if it's a mobile or home, etc. This is to make sure that the SDVC have two avenues of (realtime?) communication. 

Regarding username, personally I prefer sites that have my login credentials be my email and not an username; less to remember for me. If there are planned features that would require an username, then we'll just include it now.

**Comment by TheOnlyRealTodd** [comment]
All valid points. We also need to determine our account structure on the identity side of things. I.E. Will we have one master account type that can be set as a Veteran or as a VSO or will we have entirely 2 separate account modules independent of one another? The latter would require, for example, someone to register two separate accounts entirely if they were both a veteran and a VSO in the system, in which case we would have to provide some disambiguity mechanism for the stored email address, whereas the first option would allow one single "account" on the back-end to be either a VSO, a Veteran, or in special circumstances (site admin who needs to see things from both sides, Veteran who needs services and is also a VSO, etc...) both.

These can be handled easily with a roles-based system under one account, or have the two accounts completely independent of one another, depending on what we decide.

Another point - What if a VSO needs to have multiple representative accounts? Will the VSO account be tied to the business itself and they share the password/creds OR will we allow for a VSO "owner" user to "allow" other reps to had admin rights over the VSO account?

In the last case, we will have to have VSO accounts themselves completely abstracted out from the VSO "profile" which they will be able to "have" or "have control over." In the first case, where they just use one account and share a password (cheaper method basically), the account can **be** the VSO profile one-in-the-same.

Either way, we will have to have a claims-based system identify the userId individually obviously so each user can get into his/her account only.

**Comment by stevengum** [comment]
Modified the model files to include branch_id, city_id, vet_id fields in the models where they are foreign keys for sql operations.
